### PR TITLE
Igniters ignite shit.

### PR DESCRIPTION
**Pull request by KazeEspada**
Igniters now call fire_act() on everything in their tile. This makes them... you know... actually ignite shit?

closes #16890

### CODE REVIEW

**Comment by KazeEspada** [comment]
This detonates fuel tanks, makes them into soft stun mines, and ALL SORTS OF FUN SHIT.

**Comment by Zilenan91** [comment]
There any way for you to attach them to fuel tanks so that the tanks don't get pushed around?

**Comment by KazeEspada** [comment]
No, place your fuel tanks better.

File: code/modules/assembly/igniter.dm [hunk 1]
```diff
@@ -24,6 +24,9 @@
 	var/turf/location = get_turf(loc)
 	if(location)
 		location.hotspot_expose(1000,1000)
+		for(var/T in location.contents)
+			var/atom/A = T
+			A.fire_act()
```
Comment (phil235) [inline code/modules/assembly/igniter.dm]: This will burn (i.e. becomes a pile of ash after a while) every objects on the tile. :-1:

Comment (Robustin) [inline code/modules/assembly/igniter.dm]: Eh I reacted that way at first too Phil, but honestly its fine. It's just one tile and once people learn that igniters are actually dangerous now, it won't be hard to manage.

**Comment by Robustin** [comment]
Eh, works for me. I still think having to drop igniter assemblies, then drag the fuel tanks over the assembly, is a little clunky. It also lets you set proximity tanks which I'm kinda iffy on. On one hand part of me is like HELL YEA IMPROVISED AMBUSHES, THIS IS GONNA GIVE US SOME GREAT MOMENTS. On the other hand I absolutely feel for the sec officer who has to worry about another GGnoRE tactic that will realistically be hard to deal with in places like maint.

File: code/modules/assembly/igniter.dm (hunk 1 above)
Comment (phil235) [inline code/modules/assembly/igniter.dm]: Grab an igniter, click it in your hand, Congratulation, you're on fire!

**Comment by Supermichael777** [comment]
>not throwing a dead mouse down every tunnel to trigger mines

**Comment by phil235** [comment]
@Supermichael777 
If thrown mobs trigger mines, that's a bug.

**Comment by ChangelingRain** [comment]
Projectiles do, too.

**Comment by CosmicScientist** [comment]
I thought fuel tanks caused breaches to lattice?

File: code/modules/assembly/igniter.dm (hunk 1 above)
Comment (KazeEspada) [inline code/modules/assembly/igniter.dm]: No.. its an IGNITER not a SPARKER

**Comment by KazeEspada** [comment]
Since the only thing hiding the assembly is the fuel tank, its easy to spot.

**Comment by Zilenan91** [comment]
This could be implemented more intuitively but yeah it's good :+1:

**Comment by TechnoAlchemisto** [comment]
but igniters are basically just tiny pilot lights/lighters why would they set everything above them ablaze?  Especially a person who is just stepping over it

**Comment by Zilenan91** [comment]
Maybe NT, instead of stocking the station with actual igniters stocked it with mini flamethrowers because screaming metal death trap

**Comment by KazeEspada** [comment]
We have lighters, igniters are in a whole nother category. Hence why they are needed for the stun prod and flamethrowers.

**Comment by Dorsisdwarf** [comment]
Igniters are just spark plugs.

Source: Everywhere else they're used.

File: code/modules/assembly/igniter.dm (hunk 1 above)
Comment (xxalpha) [inline code/modules/assembly/igniter.dm]: It ignites but it doesn't hold any fuel. It should not light things on fire without fuel.

Comment (PKPenguin321) [inline code/modules/assembly/igniter.dm]: Space magic

Comment (xxalpha) [inline code/modules/assembly/igniter.dm]: You can rig an igniter to a remote, put it inside any container, give it to someone and then light them on fire from a distance.

This is just not well thought out.

Comment (PKPenguin321) [inline code/modules/assembly/igniter.dm]: Actually that's hilarious and a good example of creative emergent gameplay

Comment (xxalpha) [inline code/modules/assembly/igniter.dm]: That's not creative or emergent, it's just broken and makes no sense.

Comment (PKPenguin321) [inline code/modules/assembly/igniter.dm]: Sure it does, you have an igniter in your pocket and it goes off, lighting you and your clothes on fire
what's not to get?

**Comment by KazeEspada** [comment]
Theres no way to have this affect only fueltanks elegantly. Besides this is fine, especially since its like 2 fire stacks so its only one roll to extingush.

**Comment by xxalpha** [comment]
It makes no fucking sense to ignite everything on a turf on fire from inside something. That's not how fire works and the igniter only creates sparks, it ignites by sparking.

**Comment by Zilenan91** [comment]
It'd be a flame blast after this which I'm personally fine with because it makes improvised shit not only even more dangerous but also more unreliable.

**Comment by nullbear** [comment]
Just make it only check for non-humans.

**Comment by Razharas** [comment]
I made igniters be used in stunprods because they are actually sparkers now
So no to this

**Comment by PKPenguin321** [comment]
Aw

**Pull request closed without merging**